evaluation/gaia/run_infer.py

```diff
@@ -0,0 +1,336 @@
+        instruction += f"\n\nThe mentioned file is provided in the workspace at: {dest_file.split('/')[-1]}"
+
+    instruction += 'IMPORTANT: You should ONLY interact with the environment provided to you AND NEVER ASK FOR HUMAN HELP.\n'
+    # NOTE: You can actually set slightly different instruction for different agents
+    instruction += AGENT_CLS_TO_INST_SUFFIX.get(agent_class, '')
+
+    # Here's how you can run the agent (similar to the `main` function) and get the final task state
+    state: State = asyncio.run(
+        main(
+            instruction,
+            fake_user_response_fn=AGENT_CLS_TO_FAKE_USER_RESPONSE_FN.get(agent_class),
+            sandbox=sandbox,
+        )
+    )
+    # ======= Attempt to evaluate the agent's edits =======
+    # If you are working on simplier benchmark that only evaluates the final model output (e.g., in a MessageAction)
+    # You can simply get the LAST `MessageAction` from the returned `state.history` and parse it for evaluation.
+
+    if state is None:
+        raise ValueError('State should not be None.')
+
+    model_answer = ''
+    for act, _ in reversed(state.history):
+        if isinstance(act, MessageAction) and act.source == 'model':
+            model_answer = act.content
+            break
+    logger.info(
+        f'Final message: {model_answer} | Ground truth: {instance["Final answer"]}'
+    )
+    test_result = question_scorer(
+        model_answer=model_answer, ground_truth=instance['Final answer']
+    )
+
+    # Save the output
+    output = {
+        'instance_id': instance['task_id'],
+        'instance': instance,
+        'instruction': instance['Question'],
+        'metadata': metadata,
+        'history': [
+            (event_to_dict(action), event_to_dict(obs)) for action, obs in state.history
+        ],
+        'error': state.error if state and state.error else None,
+        'test_result': test_result,
+        'model_answer': model_answer,
+    }
+
+    # Close the sandbox
+    sandbox.close()
+    return output
+
+
+if __name__ == '__main__':
+    parser = get_parser()
+    parser.add_argument(
+        '--level',
+        type=str,
+        help='gaia level to evaluate, eg. 2023_level1',
+    )
+    parser.add_argument(
+        '--data-split',
+        type=str,
+        help='data split to evaluate, eg. validation',
+    )
+    args, _ = parser.parse_known_args()
+    if args.directory:
+        config.workspace_base = os.path.abspath(args.directory)
+        print(f'Setting workspace base to {config.workspace_base}')
+    # NOTE: It is preferable to load datasets from huggingface datasets and perform post-processing
+    # so we don't need to manage file uploading to OpenDevin's repo
+    level = args.level
+    data_split = args.data_split
+    dataset = load_dataset('gaia-benchmark/GAIA', level)
+    huggingface_hub.snapshot_download(
+        'gaia-benchmark/GAIA',
+        repo_type='dataset',
+        token='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',
```

Comment:
huh is this your token? will it cause security risks though?

Reply:
We could ask user to input their token via `input()` -- since they need to accept GAIA's ToS before able to use it

Reply:
😅 thanks for catching this. I've invalided it. Sure, we should let users accept it. I will make the change